evalml/tests/conftest.py

```diff
@@ -193,6 +193,8 @@ def assert_allowed_pipelines_equal_helper():
     def assert_allowed_pipelines_equal_helper(
         actual_allowed_pipelines, expected_allowed_pipelines
     ):
+        actual_allowed_pipelines.sort(key=lambda p: p.name)
```

Comment:
Since pipelines used to be built in `AutoMLSearch`, tests would compare against an unsorted list of pipelines. Changed due to `self.allowed_pipelines` in `IterativeAlgorithm` sorting due to `_ESTIMATOR_FAMILY_ORDER`.

Reply:
If we want to do this, we have tests in place then to confirm that the order of allowed_pipelines as as expected? My concern here is that by sorting both lists, we're able to confirm that the types of pipelines match, but not the order in which they're executed anymore.

Reply:
Yes I agree, but from what I could tell `assert_allowed_pipelines_equal_helper` isn't used in any tests that are checking for order and we have tests in `test_iterative_algorithm.py` like `test_iterative_algorithm_first_batch_order_param` that do account for the case!

Reply:
To clarify: since most of the tests in `test_automl` etc use the default iterative algorithm, the pipelines will be sorted in the order defined by `_ESTIMATOR_FAMILY_ORDER` in iterative algorithm. But since these tests directly compare against `make_pipeline`, the pipelines compared are in a different order. Another solution to this would be to sort the order into `_ESTIMATOR_FAMILY_ORDER` but I opted for a simpler solution.